### PR TITLE
Add support for direct-tcpip channels

### PR DESCRIPTION
**Pull request by kyrias**
This PR exposes the `russh` support for opening `direct-tcpip` channels.

`direct-tcpip` channels are used to implement things like local port forwards.  (Equivalent to the `ssh` `-L` flag.)

### CODE REVIEW

**Review by Miyoshi-Ryota — COMMENTED** [review]
@kyrias 
Sorry for not studying enough. I still don't understand the demand for this method.

Is it possible to add a test as an example of usage? And/or could you please add an explanation to the PR about when it is needed?

Thank you for your contribution.

Note: Don't worry about unittest failure on CI as it has been fixed in the current master.

**Comment by kyrias** [comment]
> Is it possible to add a test as an example of usage?

Sure, I'll look into it.

> And/or could you please add an explanation to the PR about when it is needed?

Is that adequate?

**Comment by Miyoshi-Ryota** [comment]
thanks. yes, it is adequate.

**Comment by kyrias** [comment]
Test pushed.

File: src/client.rs [hunk 1]
```diff
@@ -562,6 +606,31 @@ ASYNC_SSH2_TEST_UPLOAD_FILE
...
+        let mut stream = channel.into_stream();
+        stream.write_all(b"GET / HTTP/1.0\r\n\r\n").await.unwrap();
+
+        let mut response = String::new();
+        stream.read_to_string(&mut response).await.unwrap();
+
+        let body = response.split_once("\r\n\r\n").unwrap().1;
+        assert_eq!("Hello", body);
+    }
+
```
Comment (Miyoshi-Ryota) [inline src/client.rs:633]: Thank you! I can understand what `direct_tcpip_channel` is from this test. Perfect Thanks.

**Review by Miyoshi-Ryota — APPROVED** [review]
LGTM

**Pull request merged**